### PR TITLE
[Storage][DataMovement] Combine TransferManager.StartTransferAsync methods

### PR DESCRIPTION
**Pull request by jalauzon-msft**
This change combines the two `TransferManager.StartTransferAsync` methods (one for single transfer and one for container transfers) into one single method that accepts the base class of both single and container resources. The new method does type checking to decide what type of job to make.

The largest part of this change was the existing `StorageResource` was renamed to `StorageResourceSingle` so the base class, `StorageResouceBase` could be renamed to just `StorageResource`.

There is still quite a bit of shared code between the new private methods to create single versus container resources that could be refactored but that was out of scope for this change and can be handled later.

### CODE REVIEW

File: sdk/storage/Azure.Storage.DataMovement/src/Shared/TransferManager.cs [hunk 1]
```diff
@@ -375,11 +375,54 @@ internal virtual Task<bool> TryRemoveTransferAsync(string id)
...
+                destinationResource is StorageResourceSingle)
+            {
+                transferJobInternal = await BuildSingleTransferJob(
+                    (StorageResourceSingle)sourceResource,
+                    (StorageResourceSingle)destinationResource,
+                    transferOptions,
+                    dataTransfer,
+                    resumeJob).ConfigureAwait(false);
+            }
+            // Cotnainer transfer
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.DataMovement/src/Shared/TransferManager.cs:394]: NIT - container*

**Review by amnguye — COMMENTED** [review]
LGTM

**Review by amnguye — APPROVED** [review]

**Review by jaschrep-msft — APPROVED** [review]

**Pull request merged**